src/lib.rs

```diff
@@ -145,47 +145,66 @@ impl KnownHosts {
     }
 }
 
-impl Session {
-    fn ctl_path(&self) -> std::path::PathBuf {
-        self.ctl.path().join("master")
+/// Build a [`Session`] with options.
+#[derive(Debug)]
+pub struct SessionBuilder {
+    user: Option<String>,
+    port: Option<String>,
+    keyfile: Option<std::path::PathBuf>,
+    known_hosts_check: KnownHosts,
+}
+
+impl Default for SessionBuilder {
+    fn default() -> Self {
+        Self {
+            user: None,
+            port: None,
+            keyfile: None,
+            known_hosts_check: KnownHosts::Add,
+        }
     }
+}
 
-    /// Connect to the host at the given `addr` over SSH.
+impl SessionBuilder {
+    /// Set the ssh user.
     ///
-    /// The format of `destination` is the same as the `destination` argument to `ssh`. It may be
-    /// specified as either `[user@]hostname` or a URI of the form `ssh://[user@]hostname[:port]`.
+    /// Default `None`, which will follow your ssh config.
```

Comment:
I find this wording a little confusing. I think we should just have it say
> Defaults to `None`.

And then the ssh config stuff is implied by the crate-level docs.